lab-steven/server.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client-constructor.js');
+const PORT = process.env.PORT || 8080;
+const server = net.createServer();
+const ee = new EE();
+
+const connectedClients = [];
+
+ee.on('@all', function(user, data) {
+  connectedClients.forEach(function(client) {
+    client.socket.write(`${user.nickname}: ${data}\r\n`);
+  });
+});
+
+ee.on('@nickname', function(user, data) {
+  user.nickname = data.trim();
+  user.socket.write(`Changed nickname to ${data}\r\n`);
+});
+
+ee.on('@pm', function(user, data) {
+  var message = data.split(' ').slice(1).join(' ').trim();
+  var targetUser = data.split(' ').shift().trim();
+
+  var foundUser = connectedClients.filter(function(client) {
```

Comment:
You should check out the [Array.prototype.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) function for this.